src/commands/status.ts

```diff
@@ -12,27 +12,25 @@ export default class Status extends BlockchainCommand {
     const { flags } = await this.parse(Status);
     const provider = await getProvider(flags.network, flags.rpc);
     const registry = await getContract(flags.network, flags.abi, "EarthfastRegistry", provider);
+    const billing = await getContract(flags.network, flags.abi, "EarthfastBilling", provider);
     const netVersion = await registry.getVersion();
     const cliVersion = version;
 
-    // Round epoch start to Wednesday at 16:00 UTC
-    const lastEpochStart = new Date();
-    lastEpochStart.setUTCHours(16, 0, 0, 0);
-    // Get current day (0 = Sunday, 3 = Wednesday)
-    const currentDay = lastEpochStart.getUTCDay();
-    // Calculate days to subtract to reach previous Wednesday
-    const daysToSubtract = (currentDay - 3 + 7) % 7 || 7; // If result is 0, subtract 7 days
-    lastEpochStart.setUTCDate(lastEpochStart.getUTCDate() - daysToSubtract);
+    // last is current epoch
+    const lastEpochStart = Number(await registry.getLastEpochStart());
+    const lastEpochLength = Number(await registry.getLastEpochLength());
```

Comment:
Instead of directly converting the values using `Number()`, it's safer to first check if the values are valid. This approach ensures that if `registry.getLastEpochStart()` or `registry.getLastEpochLength()` return `null`, `undefined`, or other falsy values, we default to `0` instead of getting `NaN`.


```
const lastEpochStartRaw = await registry.getLastEpochStart();
const lastEpochLengthRaw = await registry.getLastEpochLength();

const lastEpochStart = lastEpochStartRaw ? Number(lastEpochStartRaw) : 0;
const lastEpochLength = lastEpochLengthRaw ? Number(lastEpochLengthRaw) : 0;
```